internal/core/items/text.rs

```diff
@@ -694,10 +694,10 @@ impl TextInput {
         anchor_pos != cursor_pos
     }
 
-    fn selected_text(self: Pin<&Self>) -> String {
+    fn with_selected_text<T>(self: Pin<&Self>, cb: impl FnOnce(&str) -> T) -> T {
```

Comment:
If it is only used once, maybe it doesn't deserve a function with such a complicated signature.